src/Goerli-DssSpellCollateral.sol

```diff
@@ -28,61 +61,257 @@ contract DssSpellCollateralAction {
+        address MCD_JOIN_DAI
+    ) internal {
+        // RWA008-A collateral deploy
+        bytes32 ilk      = "RWA008-A";
+        uint256 decimals = DSTokenAbstract(RWA008).decimals();
+
+        // Sanity checks
+        require(GemJoinAbstract(MCD_JOIN_RWA008_A).vat() == MCD_VAT,  "join-vat-not-match");
+        require(GemJoinAbstract(MCD_JOIN_RWA008_A).ilk() == ilk,      "join-ilk-not-match");
+        require(GemJoinAbstract(MCD_JOIN_RWA008_A).gem() == RWA008,   "join-gem-not-match");
+        require(GemJoinAbstract(MCD_JOIN_RWA008_A).dec() == decimals, "join-dec-not-match");
+
+        require(RwaUrnLike(RWA008_A_URN).vat() == MCD_VAT,               "urn-vat-not-match");
+        require(RwaUrnLike(RWA008_A_URN).jug() == MCD_JUG,               "urn-jug-not-match");
+        require(RwaUrnLike(RWA008_A_URN).daiJoin() == MCD_JOIN_DAI,      "urn-daijoin-not-match");
+        require(RwaUrnLike(RWA008_A_URN).gemJoin() == MCD_JOIN_RWA008_A, "urn-gemjoin-not-match");
+
+        // Init the RwaLiquidationOracle
+        RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).init(ilk, RWA008_A_INITIAL_PRICE, RWA008_DOC, RWA008_A_TAU);
+        (, address pip, , ) = RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).ilks(ilk);
+
+        // Set price feed for RWA008
+        DssExecLib.setContract(MCD_SPOT, ilk, "pip", pip);
+
+        // Init RWA008 in Vat
+        Initializable(MCD_VAT).init(ilk);
+        // Init RWA008 in Jug
+        Initializable(MCD_JUG).init(ilk);
+
+        // Allow RWA008 Join to modify Vat registry
+        DssExecLib.authorize(MCD_VAT, MCD_JOIN_RWA008_A);
+
+        // Allow RwaLiquidationOracle2 to modify Vat registry
+        DssExecLib.authorize(MCD_VAT, MIP21_LIQUIDATION_ORACLE);
+
+        // Set the debt ceiling
+        DssExecLib.increaseIlkDebtCeiling(ilk, RWA008_A_LINE, /* _global = */ true);
+
+        // Set the stability fee
+        DssExecLib.setIlkStabilityFee(ilk, RWA008_A_RATE, /* _doDrip = */ false);
+
+        // Set the collateralization ratio
+        DssExecLib.setIlkLiquidationRatio(ilk, RWA008_A_MAT);
+
+        // Poke the spotter to pull in a price
+        DssExecLib.updateCollateralPrice(ilk);
+
+        // Give the urn permissions on the join adapter
+        DssExecLib.authorize(MCD_JOIN_RWA008_A, RWA008_A_URN);
+
+        // Helper contract permisison on URN
+        RwaUrnLike(RWA008_A_URN).hope(RWA008_A_URN_CLOSE_HELPER);
+        RwaUrnLike(RWA008_A_URN).hope(RWA008_A_OPERATOR);
+
+        // Set up output conduit
+        RwaOutputConduitLike(RWA008_A_OUTPUT_CONDUIT).hope(RWA008_A_OPERATOR);
+
+        // Whitelist DIIS Group in the conduits
+        RwaOutputConduitLike(RWA008_A_OUTPUT_CONDUIT).mate(RWA008_A_MATE);
+        RwaInputConduitLike(RWA008_A_INPUT_CONDUIT)  .mate(RWA008_A_MATE);
+
+        // Whitelist Socgen in the conduits
+        RwaOutputConduitLike(RWA008_A_OUTPUT_CONDUIT).mate(RWA008_A_OPERATOR);
+        RwaInputConduitLike(RWA008_A_INPUT_CONDUIT)  .mate(RWA008_A_OPERATOR);
```

Comment:
This will give `SocGen` the ability to `push()`, but from the diagrams at https://forum.makerdao.com/t/rwa008-ofh-mip21-token-ces-domain-team-assessment/15239 it seems like the DIIS group are the only ones that are supposed to call `push()`.

```suggestion
```

Reply:
No, this is correct.
SocGen is whitelisted here as a fallback in case DIIS Group is not able to process the request.

Reply:
Okay, just making sure.